extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -234,13 +236,17 @@ export abstract class ToolBase implements ITool {
 	}
 
 	public async loadInformation(): Promise<void> {
+		await this._pendingVersionAndStatusUpdate;
 		if (this.status === ToolStatus.NotInstalled) {
 			await this.addInstallationSearchPathsToSystemPath();
-			this.status = await this.updateVersionAndGetStatus();
 		}
 	}
 
-	private async updateVersionAndGetStatus(): Promise<ToolStatus> {
+	private startVersionAndStatusUpdate() {
+		this._pendingVersionAndStatusUpdate = this.updateVersionAndStatus();
```

Comment:
Make sure you add a catch to the call to avoid the dangling promise (and handle it appropriately by setting vars here as needed so the user knows there was a problem)

Reply:
I am not sure if an additional try-catch is necessary. At least not by this change.  

If you look at the body of the method updateVersionAndStatus(), it executes async process executions. It then awaits them, and in case these async process executions do not yield the expected status, then bubbles the error through dialogue messages and Deployment output channels.

If this method throws which happens when the awaited process Execution throws, then the state is already captured in the deployment output channel by the process execution before the error was thrown from those executions. In our calling code when we await the this._pendingVersionAndStatusUpdate Promise this exception gets rethrown and bubbled up just the way it was before. Now looking closely we await on this promise (as we were doing before this change) in two locations:
1. From the install method where it is caught and handled appropriately.
2. From the loadInformation method, which is called from the updateToolsDisplayTable method in resourceTypePickerDialog.ts. There the errors bubbled up from loadInformation are not being handled. But this was the case before so I have filed the issue: 8640 (https://github.com/microsoft/azuredatastudio/issues/8640) to track this issue.

Since this was a pre-existing issue which has not impacted any known user experience thus far, we are going to evaluate and handled this issue independently and track it with https://github.com/microsoft/azuredatastudio/issues/8640

Reply:
I'm saying just add a catch to clean up the dangling promise so if an error occurs it doesn't escape this context. Regardless of whether the called method can have an error escape the compiler doesn't know that and it's best just to err on the safe side and assume that an error could be thrown and handle it appropriately.

Let me know if you want more information - I can go into further detail about why we're doing this.